src/App.tsx

```diff
@@ -14,11 +17,108 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isReversed: boolean,
+  Goods: string[],
+  initialGoods: string[],
+  listHidden: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    Goods: goodsFromServer,
+    initialGoods: goodsFromServer,
+    listHidden: true,
+  };
+
+  startButton() {
+    this.setState(prevState => ({
+      listHidden: !(prevState.listHidden),
+    }));
+  }
+
+  reverseGoods() {
+    this.setState(prevState => ({
+      isReversed: !(prevState.isReversed),
+      Goods: prevState.Goods.reverse(),
```

Comment:
Don't change state, do copy for this:
```suggestion
      Goods: [...prevState.Goods].reverse(),
```